compiler/options.nim

```diff
@@ -632,8 +632,9 @@ proc isDefined*(conf: ConfigRef; symbol: string): bool =
     of "cpu32": result = CPU[conf.target.targetCPU].bit == 32
     of "cpu64": result = CPU[conf.target.targetCPU].bit == 64
     of "nimrawsetjmp":
+      # faster on BSDs, prevents stack corruption on Windows: https://github.com/nim-lang/Nim/pull/19197
```

Comment:
Pretty confusing comments, because it prevents stack corruption only with MinGW64 C compiler version `12.1.0` which is bundled with `msys2`. But mentioned stack corruption issue could not be reproduced using other C compilers, for example version `11.2.0` is not affected by the issue. Also i'm not sure about BSD and performance because FreeBSD as well as others are now using `clang`. And i'm not sure `__builtin_setjmp` is implemented for all the platforms for `clang`.

Reply:
>  it prevents stack corruption only with MinGW64 C compiler version 12.1.0

Wrong.

> version 11.2.0 is not affected by the issue

Wrong again. This has nothing to do with compiler versions. It's all in the Mingw-w64 CRT (C runtime), as clearly explained here: https://github.com/status-im/nimbus-eth2/issues/3121

Yes, you may see the heisenbug magically go away (and come back), when using different Mingw-w64 compiler versions, but that's because the stack gets rearranged and unwinding corruption no longer hits the function call stack. But it's still there.

> i'm not sure about BSD and performance

A very old piece of documentation gives performance as the reason why rawSetJmp is the default on BSDs. I just copied the info from there. Feel free to benchmark.

> FreeBSD as well as others are now using clang

Irrelevant.

> And i'm not sure __builtin_setjmp is implemented for all the platforms for clang.

Also irrelevant. You're confusing "__builtin_setjmp" with "_setjmp", which is what we mean by "raw" here.